src/main/java/de/tum/cit/aet/artemis/programming/service/localci/SharedQueueManagementService.java

```diff
@@ -289,4 +300,125 @@ public Page<BuildJob> getFilteredFinishedBuildJobs(FinishedBuildJobPageableSearc
         return new PageImpl<>(orderedBuildJobs, buildJobIdsPage.getPageable(), buildJobIdsPage.getTotalElements());
     }
 
+    /**
+     * Estimates the queue release date for a build job based on the participation ID.
+     * This method calculates the estimated queue duration for a buildjob with the given participation ID.
+     * It takes into account the current queue state, the capacity of build agents,
+     * and the remaining duration of currently processing jobs.
+     *
+     * @param participationId the ID of the participation for which the queue release date is estimated
+     * @return the estimated queue release date as a {@link ZonedDateTime}
+     */
```

Comment:
```suggestion
    /**
     * Estimates the queue release date for a build job based on the participation ID.
     *
     * @param participationId the ID of the participation for which the queue release date is estimated
     * @return the estimated queue release date as a {@link ZonedDateTime}
     */
```

I am not sure if these details are relevant on the javadoc level, I would regard them as implementation details that do not need to be visible to the caller